### DISCUSSION TITLE
It works!!! 🎉

### DISCUSSION THREAD

**Discussion by maks** [xiphonics/picoTracker-pcb#4] [General]
hi @democloid, wanted to let you know, I finally got all the parts for the "desktop edition" and had a chance to wire everything together, tried firing it up to play the 10k project and it **WORKS!** 🎉 

https://www.youtube.com/shorts/gvZeiHrScno

As you can see still very messy, I need to properly wire up all the 3.3Vs and GNDs as I ended up going the route of using header pins after soldering the first few connections directly as I realised I might need to switch out the pico or one of the other boards if I accidentally blew one or more of them up 😄 . 

Also I've just received the PCBs from JLCPCB yesterday and they all look good, but I now need to wait for the remaining parts to make their way to me from china. I've been experimenting with printing various low profile keycaps, but same as you I haven't been too happy with the results so far so likely will end up just using XDA keycaps (once that aliexpress order gets here too). 

I have some small ideas to work on the firmware now that I have the first build done so I'll add some issues to your repo and hopefully have some PRs in the near future. Also I want to have a go at using your port as the basis of getting other ports working like for the nabernic rg280v running newer firmware where my prev attempt from the original lgpt codebase had audio issues.

Thank you so much again for creating this awesome project! I can't wait to keep going building the portable edition and then hacking on the software as well! 🙂 

ps. heres a slightly out of focus pic showing I ended up using a diff sdcard board that I happened to have and it works fine, though it did take me a bit of thinking to realise that the fritzing diagram was showing connections for sdio and that I didn't need the last couple of wires using SPI.

![PXL_20231013_092701103](https://github.com/democloid/picoTrackerPortable-pcb/assets/71999/0aa65a95-cd12-46cf-bac9-55be43386168)

**Comment by democloid** [comment]
That's great! I was just thinking about your project and wondering where it might be at.

Regarding the SD card, I started using SDIO because I thought it just made sense to get better performance. In practice tho, it didn't and had some additional issues plus more memory needed for it, so I continued using SPI. Original plan was potentially stream the samples from SD, but it's a bit hard because the Pico doesn't have any external memory controller, so the SD card is not memory mapped. I later realized that I did something wrong when doing the SDIO stuff and might be related to getting low performance, so I should go back to it.

In terms of software, sounds good! bring it on. While I haven't worked on it in the last months, I have some ideas for v1.1 including adding synth instruments and improving battery performance. I think the main issue with this port and what limits a lot of things is memory usage. A particular thing that would be nice is to reduce dynamic memory usage as much as possible, but it's all over the place, so not at easy task. Using only static memory would make development exponentially easier.

BTW, in terms of using this to fix other ports, be aware that the issue with other ports may exclusively be related to SDL, which I don't use at all here, so may not be of much help.

**Comment by maks** [comment]
Brill thanks that makes sense. I've made a [new issue for sdcard streaming](https://github.com/democloid/picoTracker/issues/2) since I think that would be a great feature to have.

And yes definitely with ports I think SDL is a big issue and its on my todo list to attempt to port to SDL2, since 1.2 is ancient at this point and I have a Odroid Go Super for instance where SDL2 is pretty much mandatory in order to be able to use their higher perf gfx stack. Its also been suggested to me in comment on YT that others have had some luck with using SDL instead of asound and jack for audio on ports to other small handheld "retro game" type devices.

Finally here is a better photo of my now fully assembled desktop picoTracker, though I have to admit I don't want to show the side view as I really struggled to get the 2 halves of the case together given my choice to use pin headers on all the boards as that definitely took up more space! 😅  I definitely see the huge improvement/benefit of going to the PCB design of the portable! 🙂 

![PXL_20231014_064155589](https://github.com/democloid/picoTrackerPortable-pcb/assets/71999/8c6c4888-fd74-4689-88b2-0c34ffd68c9f)

**Reply by democloid** [reply]
I initially thought about building a PCB on which the pico and display could be mounted and be through hole components for the rest of the things in order to make an easy build. That turned out not to be compatible with this layout and when I was iterating at some point I thought if I need to change the design and I have to use SMD parts anyway I might as well do something completely new. That's how the portable version was born.

BTW, if you want to match the colors of LGPT with the bezel, the color scheme settings are compatible with picoTracker. (The only settings that I actually carried over from the original project, in a pretty hacky way)
